scripts/codepregen.py

```diff
@@ -119,9 +119,12 @@ def main(log_level, parallel, dry_run, generator, input_glob, sdk_root, output_d
 
     if generator == 'zap':
         filter.file_type = IdlFileType.ZAP
-    elif generator == 'codegen':
+    elif generator == 'codegen' or generator == 'codegen-cpp-only':
         filter.file_type = IdlFileType.MATTER
 
+    if generator == 'codegen-cpp-only':
```

Comment:
canonical way would be `filter.cpp_only == (generator == 'codegen-cpp-only')` to guarantee this is set to true/false. However I still have the question of why to understand if this placement is appropriate or not.